airbyte-integrations/connectors/source-gitlab/source_gitlab/streams.py

```diff
@@ -311,6 +311,26 @@ class Branches(GitlabChildStream):
     flatten_id_keys = ["commit"]
     flatten_parent_id = True
 
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[StreamData]:
+        try:
+            yield from super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
+        except requests.exceptions.HTTPError as error:
+            status = error.response.status_code
+            if status == 404:
```

Comment:
we already do have handling of 403 error, could it be extended to also cope with 404?

Reply:
No, because this fix is done only for branches stream.

Reply:
We could make it customizable per stream, for instance:

```
class GitlabStream:
    non_retriable_codes: List[int] = [403]
    ...
    ...
    if response.status_code in self.non_retriable_codes:
        setattr(self, "raise_on_http_errors", False)
        self.logger.warning(
            f"Got 403 error when accessing URL {response.request.url}."
            f" Very likely the feature is disabled for this project and/or group. Please double check it, or report a bug otherwise.")
        return False
    return super().should_retry(response)
```

```
class Branches:
    non_retriable_codes = [404]